data-exporter/src/main/java/org/reactome/release/NCBIGene.java

```diff
@@ -0,0 +1,204 @@
+				continue;
+			}
+
+			for (String ncbiGeneId : ncbiEntry.getNcbiGeneIds()) {
+				ncbiGeneXMLNodeStrings.add(ncbiEntry.getEntityLinkXML(ncbiGeneId));
+
+				for (ReactomeEvent topLevelPathway : topLevelPathways) {
+					ncbiGeneXMLNodeStrings.add(ncbiEntry.getEventLinkXML(ncbiGeneId, topLevelPathway));
+				}
+			}
+
+			ncbiGeneLogger.info("Finished with " + ncbiEntry.getUniprotAccession());
+		}
+
+		int fileCount = 0;
+		int numberOfGeneXMLFiles = getNumberOfGeneXMLFiles(ncbiGeneXMLNodeStrings);
+		for (Set<String> ncbiGeneXMLNodeStringsSubSet : Utilities.splitSet(ncbiGeneXMLNodeStrings, numberOfGeneXMLFiles)) {
+			Path geneXMLFilePath = getGeneXMLFilePath(++fileCount);
+
+			deleteAndCreateFile(geneXMLFilePath);
+
+			logger.info("Generating " + geneXMLFilePath.getFileName());
+
+			appendWithNewLine(getXMLHeader(), geneXMLFilePath);
+			appendWithNewLine(getOpenRootTag(), geneXMLFilePath);
+			for (String ncbiGeneXMLNodeString : ncbiGeneXMLNodeStringsSubSet) {
+				appendWithNewLine(ncbiGeneXMLNodeString, geneXMLFilePath);
+			}
+
+			appendWithNewLine(getCloseRootTag(), geneXMLFilePath);
+		}
+
+		logger.info("Finished writing gene XML file(s)");
+	}
+
+	/**
+	 * The size (in bytes) for the set of all Link XML node strings to include in the final NCBI Gene XML files
+	 * is determined and the minimum number of files to write in order for each to be less than the pre-set limit is
+	 * returned.
+	 * The maximum size of each file is 15MB and is a limit set by NCBI for individual file uploads.
+	 * @param ncbiGeneXMLNodeStrings Set of all the Link XML nodes strings to include in the NCBI Gene XML file(s)
+	 * @return Number of NCBI Gene XML files to create (i.e. split the XML nodes amongst to be under the file size
+	 * limit)
+	 */
+	private int getNumberOfGeneXMLFiles(Set<String> ncbiGeneXMLNodeStrings) {
+		final int BYTES_TO_KILOBYTES = 1024;
+		final int KILOBYTES_TO_MEGABYTES = 1024;
+		final double MAX_FILE_SIZE_IN_MEGABYTES = 15.0;
+
+		int sizeInBytes = ncbiGeneXMLNodeStrings.stream().mapToInt(str -> str.getBytes().length).sum();
+		int sizeInMegaBytes = sizeInBytes / BYTES_TO_KILOBYTES / KILOBYTES_TO_MEGABYTES;
+
+		// The number of files is rounded up to ensure the minimum number of files is an integer
+		return (int) Math.ceil(sizeInMegaBytes / MAX_FILE_SIZE_IN_MEGABYTES);
+	}
+
+	private Path getProteinFilePath() {
+		return Paths.get(outputDir, "proteins_version" + reactomeVersion);
+	}
+
+	private Path getGeneXMLFilePath(int fileCount) {
+		String fileName = "gene_reactome" + reactomeVersion + "-" + fileCount + ".xml";
+		return Paths.get(outputDir, fileName);
+	}
+
+	private Path getGeneErrorFilePath() {
+		return Paths.get(outputDir, "geneentrez_" + reactomeVersion + ".err");
+	}
+
+	/**
+	 * Returns the XML header for the NCBI Gene file containing the Reactome entity and event base URLs
+	 * @return XML header as String
+	 */
+	public static String getXMLHeader() {
```

Comment:
It sounds like the NCBI file can't use the standard Dom XML libraries for output, maybe a little comment explaining why would be nice for future developers, lest they try to rewrite this code and run into problems that you have already experienced.

Reply:
Good point -- I will add that in.